internal/pkg/chain/traversal.go

```diff
@@ -117,3 +119,68 @@ func CollectTipsToCommonAncestor(ctx context.Context, store TipSetProvider, oldH
 	newTips, err = CollectTipSetsOfHeightAtLeast(ctx, newIter, commonHeight+1)
 	return
 }
+
+// ErrNoCommonAncestor is returned when two chains assumed to have a common ancestor do not.
+var ErrNoCommonAncestor = errors.New("no common ancestor")
```

Comment:
Changes here are just a vanilla move from the deleted ancestors file. No need to review (ditto for the tests).